src/core/utils.dart

```diff
@@ -1 +1,184 @@
+  print(greenColor + "Note added successfully." + resetColor);
+  }
+
+void viewNotes() {
+    print(notes.toString());
+  }
+
+void editNote(){
+  print("Choose note you want to edit.");
+  print("Do you want to choose by" + blueColor + " ID/title (1/2)" + resetColor + ": ");
+  String? input = stdin.readLineSync();
+  int searchBy = int.parse(input!);
+  switch(searchBy){
+    case 1:
+      print("enter ID: ");
+      String? input = stdin.readLineSync();
+      int byID = int.parse(input!);
+      var foundNote = notes.notes.firstWhere((note) => note.id == byID, orElse: () => NoteModel(id: -1, title: "", theNote: ""));
+      if (foundNote.id != -1) {
+        print("Note found. Enter your edit: ");
+        String? input = stdin.readLineSync();
+        foundNote.updateNote(input!);
+        print(greenColor + "Note with ID \"${foundNote.id}\" edited successfully." + resetColor);
+      } else {
+        print(redColor + "Note not found." + resetColor);
+      }
+      break;
+    case 2:
+      print("enter title: ");
+      String? byTitle = stdin.readLineSync();
+      var foundNote = notes.notes.firstWhere((note) => note.title == byTitle, orElse: () => NoteModel(id: -1, title: "", theNote: ""));
+      if (foundNote.title != "") {
+        print("Note found. Enter your edit: ");
+        String? input = stdin.readLineSync();
+        foundNote.updateNote(input!);
+        print(greenColor + "Note with title \"${foundNote.title}\" edited successfully." + resetColor);
+      } else {
+        print(redColor + "Note not found." + resetColor);
+      }
+      break;
+    default:
+      print(redColor + "Invalid choive." + resetColor);
+      break;
+  }
+}
+
+void deleteNote(){
+  print("Choose note you want to delete.");
+  print("Do you want to choose by" + blueColor + " ID/title (1/2)" + resetColor + ": ");
+  String? input = stdin.readLineSync();
+  int searchBy = int.parse(input!);
+  switch(searchBy){
+    case 1:
+      print("enter ID: ");
+      String? input = stdin.readLineSync();
+      int byID = int.parse(input!);
+      var foundNote = notes.notes.firstWhere((note) => note.id == byID, orElse: () => NoteModel(id: -1, title: "", theNote: ""));
+      if (foundNote.id != -1) {
+        notes.deleteNoteById(foundNote.id);
+        print(greenColor + "Note with ID \"${foundNote.id}\" found deleted successfully." + resetColor);
+      } else {
+        print(redColor + "Note not found." + resetColor);
+      }
+      break;
+    case 2:
+      print("enter title: ");
+      String? byTitle = stdin.readLineSync();
+      var foundNote = notes.notes.firstWhere((note) => note.title == byTitle, orElse: () => NoteModel(id: -1, title: "", theNote: ""));
+      if (foundNote.title != "") {
+        notes.deleteNoteByTitle(foundNote.title);
+        print(greenColor + "Note with Title \"${foundNote.title}\" found deleted successfully." + resetColor);
+      } else {
+        print(redColor + "Note not found." + resetColor);

```

Comment:
Instead of using global variables for colors, consider using a helper function to apply colors when printing messages. This can make your code more modular and easier to maintain.

```dart
String applyColor(String message, String color) {
  return '$color$message$resetColor';
}

// Usage
print(applyColor("Invalid choice.", redColor));